### ISSUE TITLE
[Feature Request] Ability to update specific custom fields through the API

### ISSUE THREAD

**Issue by TytoCapensis** [TheHive-Project/TheHive#1839]
### Request Type

Feature Request

### Idea

Except if I am mistaken, as today, there is no way to programatically update a single (or several) custom fields of a case through the API: we are always forced to update the *full* array of `customFields` wherever we want to add data in custom fields.

Therefore, I am suggesting to implement the ability to be able to select specific custom fields we want to update through API requests. (Note that it is indeed related to the raw API itself, not directly to TheHive4py)

### Why?

Let us consider a TheHive instance that runs in a heavily-automated environment, using the most features available (webhooks, analyzers, responders...) to run automation around the instance, and which relies on custom fields to store data related to the case's state.

This can be problematic in this setting, because it leads to two issues:

#### 1. Race condition

Let us say that, at **t=0**, a case is created with some custom fields in it. This case creation triggers a webhook, which triggers two automated processes, **A** and **B**. Both processes get a copy of the case (with the `get_case()` function of thehive4py, for example) to have its data to work with. (Note that these processes can be a standalone script, an analyzer, or a responder: it does not matter)

At **t=1**, the process **A** finishes, and has new data to write in a custom field. It updates the case with the new data in the custom field.

At **t=2**, the process **B** finishes, and has new data to write in another custom field. It updates the case with the new data in the other custom field.

However, the process **B** was still working with data in the state of **t=0**, and is forced to update the full array of custom fields with its new data, meaning that doing so will **erase the information provided by the process A**.

#### 2. No way to fine-tune webhooks

When a webhook (`CaseUpdate`) is triggered, the changes that were made to the case are highlighted (in the `details` array of the JSON provided). However, since we are forced to update all the custom fields at once, there is no way to watch over the change of a specific custom field, which makes automation based on this impossible (except if a system to monitor custom fields state is made in the background, but that would be a bit overkill and not really clean).

An example (following an API call of `update_case(case, fields=['customFields'])`) :

```
{
    "base": true,
    "details": {
        "customFields": {
            "category": {
                "order": 1,
                "string": "category_1"
            },
            "organization": {
                "order": 2,
                "string": "IT"
            }
            "even more fields": {
                "order": 3,
                "string": "and even more values..."
            }
        }
    },
    "object": {
        "_id": "(my case ID)",
        "_parent": null,
(...)
```

### How?

Technically, that functionality is already implemented in TheHive, since this is what the web interface is doing when we update a custom field manually:

```
{
    "base": true,
    "details": {
        "customFields": {
            "category": {
                "string": "category_1"
            }
        }
    },
    "object": {
        "_id": "(my case ID)",
        "_parent": null,
(...)
```

So this functionality probably "just" has to be made reachable from an endpoint.

**Comment by nadouani** [comment]
About about just calling

```
PATCH /case/CASE_ID {"customFields.category.string": "NEW_VALUE"}
```

**Comment by TytoCapensis** [comment]
You are right, after testing this syntax, it is working. Sometimes the solution is simpler than we think!

I could also see that updating several custom fields this way was working, for example when providing `{"customFields.category.string": "NEW_VALUE", "customFields.category.order": 0, "customFields.type.string": "MY_TYPE"}`

Thank you for the hint, I will probably submit a PR on thehive4py for a function that updates selected custom fields in the upcoming days/weeks.

**However,** I could notice that, when updating several values this way, only one of them (not always the first or last one in the JSON) is shown in the case's live stream. So to take again the upper example, in the live stream on the right of the case, we would only see
`customFields: {"type":{"string":"MY_TYPE"}}`, despite all the other values are correctly updated in the custom fields.

**Comment by nadouani** [comment]
@TytoCapensis can you please share a screenshot of what is displayed by the live stream when you update multiple custom fields at once?

The live stream is implemented to take into account the actions resulting from the UI, so it doesn't take into account the multi update, but can be improved.

**Comment by TytoCapensis** [comment]
Here are examples:

```
{"customFields.site.string": "AAA", "customFields.processing.string": "Ongoing"}
```
![1](https://user-images.githubusercontent.com/77985362/111965635-3d4d5c00-8af6-11eb-9a0f-10da0ce60365.png)

```
{"customFields.reviewed.string": "OK", "customFields.site.string": "ZZZ", "customFields.processing.string": "Failed"}
```
![2](https://user-images.githubusercontent.com/77985362/111965639-3de5f280-8af6-11eb-8450-5269cf0bbcbf.png)

It should be noted that I noticed other occurences where what was displayed on the live stream was *not* the first property in the JSON.
